pkg/api/topology.go

```diff
@@ -3,10 +3,62 @@ package api
 import (
 	"net/http"
 
+	"github.com/flanksource/canary-checker/pkg/db"
 	"github.com/flanksource/canary-checker/pkg/topology"
+	"github.com/flanksource/duty/models"
 	"github.com/labstack/echo/v4"
 )
 
+type Tag struct {
+	Key string `json:"key"`
+	Val string `json:"val"`
+}
+
+type TopologyRes struct {
+	componentIDs   []string          `json:"-"`
+	Components     models.Components `json:"components,omitempty"`
+	HealthStatuses []string          `json:"healthStatuses,omitempty"`
+	Teams          []string          `json:"teams,omitempty"`
+	Tags           []Tag             `json:"tags,omitempty"`
+	Types          []string          `json:"types,omitempty"`
+}
+
+func (t *TopologyRes) AddHealthStatuses(s string) {
+	for i := range t.HealthStatuses {
+		if t.HealthStatuses[i] == s {
+			return
+		}
+	}
+
+	t.HealthStatuses = append(t.HealthStatuses, s)
+}
+
+func (t *TopologyRes) AddType(typ string) {
+	for i := range t.Types {
+		if t.Types[i] == typ {
+			return
+		}
+	}
+
+	t.Types = append(t.Types, typ)
+}
+
+func (t *TopologyRes) AddTag(typ map[string]string) {
+	for k, v := range typ {
+		var exists bool
+		for _, tag := range t.Tags {
+			if tag.Key == k && tag.Val == v {
+				exists = true
+				break
+			}
+		}
+
+		if !exists {
+			t.Tags = append(t.Tags, Tag{Key: k, Val: v})
+		}
+	}
+}
```

Comment:
It would be faster if you just append tags and types and dedup them later in populate function

Right now, we will be iterating all the tags and all the types for all the components.